railties/test/application/configuration_test.rb

```diff
@@ -281,6 +281,20 @@ def change
       assert_instance_of Pathname, Rails.root
     end
 
+    test "application url is set from config file, available as Rails.application.url and sets default_url_options" do
+      add_to_config <<-RUBY
+        config.application_url = "https://rails.test"
+      RUBY
+
+      app "development"
+
+      assert_equal "https://rails.test", Rails.configuration.application_url
+      assert_instance_of ActionDispatch::Http::URL, Rails.application.url
+      assert_equal "https://rails.test", Rails.application.url.to_s
+      assert_equal({ host: "rails.test", protocol: "https", port: 443 }, Rails.application.default_url_options)
+      assert_equal({ host: "rails.test", protocol: "https", port: 443 }, Rails.application.config.action_mailer.default_url_options)
```

Comment:
I'm wondering how this list relates to `ApplicationController.renderer.defaults`. 

Would/should that also be affected?

```suggestion
      assert_equal({ host: "rails.test", protocol: "https", port: 443 }, Rails.application.config.action_mailer.default_url_options)
      assert_equal({ http_host: "rails.test", https: true }, ApplicationController.renderer.defaults)
Write
```

Reply:
Every time i've used a controller renderer outside of the request/response cycle it's never appeared to inherit the default url options from anywhere they're defined in the app.  I've always had to manually assign 'http_host' and 'https'.  Would be awesome if this PR addressed that!

Reply:
That does not work yet, I have never worked with ApplicationController.renderer.defaults. How did you use it, what was the problem and how did you solve it?

Reply:
Hi! I **just** noticed, that this seems to have been addressed [just a few months ago](https://github.com/rails/rails/commit/93038baad70d477f6cbb06fa9d48c8e429fb192f).

The problem was, for me, that 

```ruby
ApplicationController.renderer.render inline: "<%= image_tag user.avatar %>"
```

would not inherit the host and protocol from `Rails.application.default_url_options` when rendered outside of a request cycle. 

A solution, so far, had been to add this in an initializer:

```ruby
ActiveSupport::Reloader.to_prepare do
  default_url_options = Rails.application.config.action_mailer.default_url_options

  ApplicationController.renderer.defaults.merge!(
    http_host: default_url_options[:host],
    https: !default_url_options[:protocol].to_s.start_with?('http:')
  )
end
```

I just shrugged the problem off as a code architecture problem and was hoping this PR here would alleviate my confusion by unifying the entire setup – but now I understood that the underlying issue was a bug elsewhere.

Tests have obviously been added in the fix, so we might skip them here. Out of curiosity and confusion, I was wondering if the test I suggested here would be green or not :) Anyway, thank you for working on this! It's greatly appreciated.